### PR TITLE
Add BindingConfigDirectory property to BindingConfiguration

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**

### CODE REVIEW

File: src/Integration/NewConnectedMode/ConfigurationProvider.cs [hunk 1]
```diff
@@ -82,9 +83,14 @@ private BindingConfiguration TryGetBindingConfiguration(string bindingPath, Sona
...
 
-            return boundProject == null
-                ? null
-                : BindingConfiguration.CreateBoundConfiguration(boundProject, sonarLintMode);
+            if (boundProject == null)
+            {
+                return null;
+            }
+
+            var bindingConfigDirectory = Path.GetDirectoryName(bindingPath);
```
Comment (rita-g-sonarsource) [inline src/Integration/NewConnectedMode/ConfigurationProvider.cs:91]: This is the major change

**Comment by rita-g-sonarsource** [comment]
@duncanp-sonar I've added the directory property to BindingConfiguration, so it could be used to generate the file name(s) in IBindingConfigProvider implementations.

File: src/Core/Binding/BindingConfiguration.cs [hunk 2]
```diff
@@ -24,28 +24,31 @@ namespace SonarLint.VisualStudio.Core.Binding
 {
     public sealed class BindingConfiguration : IEquatable<BindingConfiguration>
     {
-        public static readonly BindingConfiguration Standalone = new BindingConfiguration(null, SonarLintMode.Standalone);
+        public static readonly BindingConfiguration Standalone = new BindingConfiguration(null, SonarLintMode.Standalone, null);
 
-        public static BindingConfiguration CreateBoundConfiguration(BoundSonarQubeProject project, SonarLintMode sonarLintMode)
+        public static BindingConfiguration CreateBoundConfiguration(BoundSonarQubeProject project, SonarLintMode sonarLintMode, string BindingConfigDirectory)
```
Comment (duncanp-sonar) [inline src/Core/Binding/BindingConfiguration.cs:29]: Casing - "bindingConfigDirectory"

File: src/Core/Binding/BindingConfiguration.cs [hunk 3]
```diff
@@ -24,28 +24,31 @@ namespace SonarLint.VisualStudio.Core.Binding
...
         {
             if (project == null)
             {
                 throw new ArgumentNullException(nameof(project));
             }
 
-            return new BindingConfiguration(project, sonarLintMode);
+            return new BindingConfiguration(project, sonarLintMode, BindingConfigDirectory);
```
Comment (duncanp-sonar) [inline src/Core/Binding/BindingConfiguration.cs:36]: Is it valid for the config path to be null?  Is there a reason for not checking it?

Comment (rita-g-sonarsource) [reply src/Core/Binding/BindingConfiguration.cs]: I didn't want to change many tests. Fixed.

File: src/Integration.UnitTests/NewConnectedMode/BindingConfigurationTests.cs [hunk 4]
```diff
@@ -40,17 +40,29 @@ public void Static_CheckStandaloneSingleton()
...
         {
             // Arrange
-            Action act = () => BindingConfiguration.CreateBoundConfiguration(null, SonarLintMode.Connected);
+            Action act = () => BindingConfiguration.CreateBoundConfiguration(null, SonarLintMode.Connected, "c:\\");
 
             // Act & Assert
             act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("project");
         }
 
+        [DataTestMethod]
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/NewConnectedMode/BindingConfigurationTests.cs:49]: I'm not sure why the `DataTestMethod` attribute exists; all of the existing data-driven tests just use `TestMethod` and they work fine...

Comment (rita-g-sonarsource) [reply src/Integration.UnitTests/NewConnectedMode/BindingConfigurationTests.cs:49]: Ah, I didn't know that, I thought you had to use DataTestMethod.

**Pull request merged**